File: test/domain/text_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:kac_gun_oldu/domain/text.dart';

void main() {
  group('capitalizeTr', () {
    test('capitalizes plain letters', () {
      expect(capitalizeTr('çamaşır yıkadım'), 'Çamaşır yıkadım');
    });

    test('turns a leading "i" into dotted "İ", not ASCII "I"', () {
      // The whole reason this helper exists: Dart's toUpperCase is
      // locale-insensitive and would produce "Ilaç".
      expect(capitalizeTr('ilaç aldım'), 'İlaç aldım');
      expect('ilaç'.toUpperCase().startsWith('İ'), isFalse);
    });

    test('turns a leading dotless "ı" into the dotless capital "I"', () {
      expect(capitalizeTr('ışık yaktım'), 'Işık yaktım');
    });

    test('is a no-op on empty input', () {
      expect(capitalizeTr(''), '');
    });

    test('leaves the rest of the sentence untouched', () {
      expect(capitalizeTr('diş hekimine gittim'), 'Diş hekimine gittim');
    });
  });

  group('upperTr', () {
    test('uppercases with Turkish i/ı rules', () {
      expect(upperTr('ne zaman yaptın?'), 'NE ZAMAN YAPTIN?');
      expect(upperTr('ilaç'), 'İLAÇ');
    });

    test('handles the month abbreviations used by the timeline gutter', () {
      expect(upperTr('Eyl'), 'EYL');
      expect(upperTr('Şub'), 'ŞUB');
    });
  });
}

File: src/domain/__tests__/text.test.ts
import { capitalizeTr } from '../text';

describe('capitalizeTr', () => {
  it('capitalizes plain ASCII letters', () => {
    expect(capitalizeTr('çamaşır yıkadım')).toBe('Çamaşır yıkadım');
  });

  it('turns a leading "i" into dotted "İ", not ASCII "I"', () => {
    expect(capitalizeTr('ilaç aldım')).toBe('İlaç aldım');
  });

  it('leaves a leading dotless "ı" as-is other than casing', () => {
    expect(capitalizeTr('ışık yaktım')).toBe('Işık yaktım');
  });

  it('is a no-op on empty input', () => {
    expect(capitalizeTr('')).toBe('');
  });
});
